src/core/util/util.js

```diff
@@ -526,6 +526,34 @@ function UtilFactory($document, $timeout, $compile, $rootScope, $$mdAnimate, $in
       }
     },
 
+    /**
+     * Get an element direct children matching a given tag name
+     *
+     * @param element Element to start walking the DOM from
+     * @param tagName Tag name to match
+     */
+    getChildren: function getChildren(element, tagName) {
+      var upperCasedTagName = tagName.toUpperCase();
+      if (element instanceof angular.element) element = element[0];
+      return Array.prototype.filter.call(element.children, function(node) {
+        return node.tagName === upperCasedTagName;
+      });
+    },
+
+    /**
+     * Get an element siblings matching a given tag name
+     *
+     * @param el Element to start walking the DOM from
+     * @param tagName Tag name to match
+     */
+    getSiblings: function getSiblings(element, tagName) {
+      var upperCasedTagName = tagName.toUpperCase();
+      if (element instanceof angular.element) element = element[0];
+      return Array.prototype.filter.call(element.parentNode.children, function(node) {
+        return element !== node && node.tagName === upperCasedTagName;
```

Comment:
Could you call `toUpperCase` on the `node.tagName` as well? In certain situations the tagName/nodeName can be lowercase (see #9509).